x-pack/plugins/embeddable_enhanced/public/drilldowns/url_drilldown/url_drilldown.tsx

```diff
@@ -0,0 +1,156 @@
+
+  public readonly order = 8;
+
+  readonly minimalLicense = 'gold';
+
+  public readonly getDisplayName = () => 'Go to URL';
+
+  public readonly euiIcon = 'link';
+
+  supportedTriggers(): UrlTrigger[] {
+    return [VALUE_CLICK_TRIGGER, SELECT_RANGE_TRIGGER];
+  }
+
+  private readonly ReactCollectConfig: React.FC<CollectConfigProps> = ({
+    config,
+    onConfig,
+    context,
+  }) => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const scope = React.useMemo(
+      () =>
+        urlDrilldownBuildScope({
+          globalScope: this.deps.getGlobalScope(),
+          contextScope: getContextScope(context),
+          eventScope: getMockEventScope(context.triggers),
+        }),
+      [context]
+    );
+
+    return (
+      <UrlDrilldownCollectConfig
+        config={config}
+        onConfig={onConfig}
+        scope={scope}
+        syntaxHelpDocsLink={this.deps.getSyntaxHelpDocsLink()}
+      />
+    );
+  };
+
+  public readonly CollectConfig = reactToUiComponent(this.ReactCollectConfig);
+
+  public readonly createConfig = () => ({
+    url: { template: '' },
+    openInNewTab: false,
+  });
+
+  public readonly isConfigValid = (
+    config: Config,
+    context: ActionFactoryContext
+  ): config is Config => {
+    const { isValid } = urlDrilldownValidateUrlTemplate(
+      config.url,
+      urlDrilldownBuildScope({
+        globalScope: this.deps.getGlobalScope(),
+        contextScope: getContextScope(context),
+        eventScope: getMockEventScope(context.triggers),
+      })
+    );
+    return isValid;
+  };
+
+  public readonly isCompatible = async (config: Config, context: ActionContext) => {
+    const { isValid, error } = urlDrilldownValidateUrlTemplate(
+      config.url,
+      urlDrilldownBuildScope({
+        globalScope: this.deps.getGlobalScope(),
+        contextScope: getContextScope(context),
+        eventScope: await getEventScope(context, this.deps),
+      })
+    );
+
+    if (!isValid) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `UrlDrilldown [${config.url.template}] is not valid. Error [${error}]. Skipping execution.`
+      );
+    }
+
+    return Promise.resolve(isValid);
+  };
+
+  public readonly getHref = async (config: Config, context: ActionContext) => {
+    return urlDrilldownCompileUrl(
+      config.url.template,
+      urlDrilldownBuildScope({
+        globalScope: this.deps.getGlobalScope(),
+        contextScope: getContextScope(context),
+        eventScope: await getEventScope(context, this.deps),
+      })
+    );
+  };
+
+  public readonly execute = async (config: Config, context: ActionContext) => {
+    const url = await urlDrilldownCompileUrl(
+      config.url.template,
+      urlDrilldownBuildScope({
+        globalScope: this.deps.getGlobalScope(),
+        contextScope: getContextScope(context),
+        eventScope: await getEventScope(context, this.deps, { allowPrompts: true }),
+      })
```

Comment:
Maybe a helper like `this.buildScopes` could be handy here.

```suggestion
    const scopes = this.buildScope(context, {allowPrompts: true});
    const url = await urlDrilldownCompileUrl(config.url.template, scopes);
```

Calling it "scopes" above as it constructs 3 scopes.

Reply:
Improved with helper, but living `scope` singular. I'd look at it not like multiple scopes, but 1 result scope that consists of different sources